.github/workflows/quality-control.yml

```diff
@@ -1,13 +1,78 @@
-name: Build
+name: Quality Control # should probably be renamed
 on:
-  push:
-    branches:
-      - main
   pull_request:
-    types: [opened, synchronize, reopened]
+    types: [ opened, synchronize, reopened ]
 jobs:
   build:
-    name: Build
+    name: CMake Build
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        compiler: [ clang-12, clang-13, clang-14, gcc-11 ]
+        build-type: [ Debug, Release ]
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          submodules: 'true'
+          fetch-depth: 0
```

Comment:
If `submodules: true`, but `fetch-depth: 0`, then fetching submodules should not have an effect at all, since with a fetch depth of 0, only the main repo will get fetched or am I mistaken?

Reply:
I think the commands work independent of each other 🤔
`fetch-depth: 0` should get all branches and tags with full commit history

submodules should work independently as per the description of the action:

> # Whether to checkout submodules: `true` to checkout submodules or `recursive` to
>     # recursively checkout submodules.
>     #
>     # When the `ssh-key` input is not provided, SSH URLs beginning with
>     # `git@github.com:` are converted to HTTPS.
>     #
>     # Default: false
>     submodules: ''

Reply:
I see. If using git one can specify the maximum nesting level of submodules that get fetched, I thought that this would be the same here.

Reply:
See here for their descriptions: https://github.com/actions/checkout